source/extensions/filters/http/ext_proc/ext_proc.cc

```diff
@@ -434,6 +434,16 @@ FilterTrailersStatus Filter::onTrailers(ProcessorState& state, Http::HeaderMap&
     return FilterTrailersStatus::StopIteration;
   }
 
+  switch (openStream()) {
```

Comment:
Need to open the gRPC stream here since the below sendBufferedBody() call also need to use the stream.

Reply:
I think doing this unconditionally here could result in unnecessary gRPC stream creation in the case where ext_proc config has disabled all calls (i.e. set headers and trailers mode to `SKIP` and body mode to `NONE`).

Although it's a little longer and uglier, the solution to this would be to only `openStream()` when a stream is needed, which would be (1) inside the `if (!body_delivered && state.bodyMode() == ProcessingMode::BUFFERED)` clause immediately before `sendBufferedData(...)`, and also at the previous location after the  `if (!state.sendTrailers())` check.

Alternatively, you could add the `openStream()` call inside of `Filter::sendBufferedData()`, not sure which is better.

Reply:
Good idea! Changed into after if (!body_delivered ... ) follow the style that gPRC stream is opened before actual data sending.